concent_api/api_testing_common.py

```diff
@@ -0,0 +1,187 @@
+        else:
+            print('SEND ({})'.format(url))
+            print('MESSAGE:')
+            print_golem_message(data)
+
+    assert all(value not in ['', None] for value in [endpoint, host, headers])
+    url         = "{}/api/v1/{}/".format(host, endpoint)
+
+    _print_data(data, url)
+
+    response    = requests.post("{}".format(url), headers = headers, data = _prepare_data(data))
+
+    _print_response(private_key, public_key, response)
+    validate_response_status(response.status_code, expected_status)
+    validate_response_message(response.content, expected_message, private_key, public_key,)
+    print()
+
+
+def _print_response(private_key, public_key, response):
+    if response.content is None:
+        print('RAW RESPONSE: Reponse content is None')
+    elif len(response.content) != 0:
+        _print_message_from_response(private_key, public_key, response)
+    else:
+        print('STATUS: {} {}'.format(response.status_code, http.client.responses[response.status_code]))
+        if response.text not in ['', None]:
+            print('RAW RESPONSE: {}'.format(response.text))
+
+
+def _print_message_from_response(private_key, public_key, response):
+    print('STATUS: {} {}'.format(response.status_code, http.client.responses[response.status_code]))
+    print('MESSAGE:')
+    print('Concent-Golem-Messages-Version = {}'.format(response.headers['concent-golem-messages-version']))
+    if response.headers['Content-Type'] == 'application/octet-stream':
+        _print_message_from_stream(private_key, public_key, response.content)
+    elif response.headers['Content-Type'] == 'application/json':
+        _print_message_from_json(response)
+    else:
+        print('RAW RESPONSE: Unexpected content-type of response message')
+
+
+def _print_message_from_json(response):
+    try:
+        print(response.json())
+    except json.decoder.JSONDecodeError:
+        print('RAW RESPONSE: Failed to decode response content')
+
+
+def _print_message_from_stream(private_key, public_key, content):
+    decoded_response = try_to_decode_golem_message(private_key, public_key, content)
+    print_golem_message(decoded_response)
+
+
+def try_to_decode_golem_message(private_key, public_key, content):
+    try:
+        decoded_response = load(
+            content,
+            private_key,
+            public_key,
+            check_time = False
+        )
+    except MessageError:
+        print("Failed to decode a Golem Message.")
+        raise
+    return decoded_response
+
+
+def timestamp_to_isoformat(timestamp):
+    return datetime.datetime.fromtimestamp(timestamp).isoformat(' ')
+
+
+def parse_command_line(command_line):
+    if len(command_line) <= 1:
+        sys.exit('Not enough arguments')
+
+    if len(command_line) >= 3:
+        sys.exit('Too many arguments')
+
+    cluster_url = command_line[1]
+    return cluster_url
+
+
+def create_task_to_compute(current_time, task_id, deadline_offset=60):
```

Comment:
There is a global constant called `DEADLINE_OFFSET`. Why is it hard-coded to 60 here instead of coming from the constant?

Reply:
changed to constant